src/node/rpc/test/membervoting_test.cpp

```diff
@@ -1008,6 +1010,308 @@ TEST_CASE("Add user via proposed call")
+    CHECK(operator_vote->second == vote_for);
+
+    const auto proposer_vote = votes.find(1);
+    CHECK(proposer_vote != votes.end());
+    CHECK(proposer_vote->second == vote_for);
+
+    const auto voter_vote = votes.find(2);
+    CHECK(voter_vote != votes.end());
+    CHECK(voter_vote->second == vote_for);
+  }
+}
+
+TEST_CASE("Passing operator vote")
+{
+  NetworkTables network;
+  GenesisGenerator gen(network);
+  gen.init_values();
+  auto new_kp = tls::make_key_pair();
+  auto new_ca = new_kp->self_sign("CN=new node");
+  NodeInfo ni = {"", "", "", "", new_ca, {}};
+  gen.add_node(ni);
+
+  // Operating member, as set in operator_gov.lua
+  const auto operator_cert = get_cert_data(0, kp);
+  const auto operator_id = gen.add_member(operator_cert, MemberStatus::ACTIVE);
+
+  // Non-operating members
+  std::map<size_t, ccf::Cert> members;
+  for (size_t i = 1; i < 4; i++)
+  {
+    auto cert = get_cert_data(i, kp);
+    members[gen.add_member(cert, MemberStatus::ACTIVE)] = cert;
+  }
+
+  set_whitelists(gen);
+  gen.set_gov_scripts(
+    lua::Interpreter().invoke<json>(operator_gov_script_file));
+  gen.finalize();
+
+  StubNodeState node;
+  MemberCallRpcFrontend frontend(network, node);
+
+  size_t proposal_id;
+
+  const ccf::Script vote_for("return true");
+  const ccf::Script vote_against("return false");
+
+  auto node_id = 0;
+  {
+    INFO("Check node exists with status pending");
+    Store::Tx tx;
+    auto read_values_j =
+      create_json_req(read_params<int>(node_id, Tables::NODES), "read");
+    ccf::SignedReq sr(read_values_j);
+
+    enclave::RPCContext rpc_ctx(operator_id, operator_cert);
+    Response<NodeInfo> r =
+      frontend.process_json(rpc_ctx, tx, operator_id, read_values_j, sr).value();
+    CHECK(r.result.status == NodeStatus::PENDING);
+  }
+
+  {
+    INFO("Operator proposes and votes for node");
+    Script proposal(R"xxx(
+      local tables, node_id = ...
+      return Calls:call("accept_node", node_id)
+    )xxx");
+
+    json proposej = create_json_req(Propose::In{proposal, node_id, vote_for}, "propose");
+    ccf::SignedReq sr(proposej);
+
+    Store::Tx tx;
+    enclave::RPCContext rpc_ctx(operator_id, operator_cert);
+    Response<Propose::Out> r =
+      frontend.process_json(rpc_ctx, tx, operator_id, proposej, sr).value();
+
+    CHECK(r.result.completed);
+    proposal_id = r.result.id;
+  }
+
+  {
+    INFO("Read current votes");
+
+    const auto readj = create_json_req_signed(
+      read_params(proposal_id, Tables::PROPOSALS), "read", kp);
+
+    Store::Tx tx;
+    enclave::RPCContext rpc_ctx(1, members[1]);
+    const Response<Proposal> proposal =
+      get_proposal(rpc_ctx, frontend, proposal_id, 1);
+
+    const auto& votes = proposal.result.votes;
+    CHECK(votes.size() == 1);
+
+    const auto proposer_vote = votes.find(operator_id);
+    CHECK(proposer_vote != votes.end());
+    // TODO: find out why this doesn't work
+    //CHECK(proposer_vote->second == vote_for);
```

Comment:
What's the current value of `proposer_vote->second`? I'm surprised it doesn't work either.

Reply:
It does work, there was a bug in the test :)